src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -100,6 +101,7 @@ public Descriptor(Version version, File directory, String ksname, String cfname,
     {
         assert version != null && directory != null && ksname != null && cfname != null && formatType.info.getLatestVersion().getClass().equals(version.getClass());
         this.version = version;
+        // toCanonical() returns a new instance most of the time in C*. Plugable FS may behave differently to reuse instance if nothing to canonicalize.
```

Comment:
Nit: I'm unsure how useful that comment is, here, for someone that doesn't have the context of the issue we're solving here. I wonder if it's not more distracting. And `File#toCanonical` is just:
```
Path canonical = PathUtils.toCanonicalPath(toPath());
return canonical == path ? this : new File(canonical);
```
so one doesn't have to go far to see that there is circumstance where no new instances may be created, if one is explicitly wondering.

Reply:
good point, removed